# Twitter Sentiment Analysis with Python and Machine Learning

![Twitter a platform for Big Data](/img/big.JPG)


>## Syed Owais Chishti
## Data Scientist

** Hello everyone, and welcome to another tutorial.
In this tutorial we learn about Twitter Sentiment Analysis. Before this we have learnt sentiment analysis for Facebook posts and comments, link [Facebook Sentiment Analysis](https://github.com/elacsoft/DataScience). **

## Curious about what we are going to learn? 
** Let's make some key points then, **
* Create an app in [Twitter Apps](https://apps.twitter.com).
* Python code to get tweets.
* Pre-Processing or cleaning of Tweets.
* Sentiment analysis using TextBlob. (You can use any API other than textblob ofcourse).

## Ready to learn further, Let's dive into it !

** First, to access Twitter API you have to make an account on [Twitter](https://twitter.com). After this go [Twitter Apps](https://apps.twitter.com). **

## But how to create an App on Twitter Apps, simply watch my video on Youtube, in which I have explained how to create an App and which four keys are required. [LinkToVideo](https://youtu.be/RTTkTq18vcA)

** The Keys we required are as follows: **
* Consumer Key (API Key)
* Consumer Secret (API Secret)
* Access Token
* Access Token Secret

## ! These keys are secret DONOT share them with anyone, Don't worry about me, I'll delete them by you see video and this notebook.
![yeah](/img/kid.JPG)

** I hope you've watched the video. Now, as you have the keys of your App. Let's code in Python.
I am using Anaconda Enviroment, Python 3.6.x. Check previous tutorials to get link to download Anaconda.
For this purpose we use Python Library called "tweepy".
To install tweepy simply open command prompt in windows or terminal in Linux/MAC and type, **

.>>> pip install tweepy

** After installing it, let's do some Python Programming. **

In [7]:
# Importing Libraries
from tweepy import Stream
from tweepy import OAuthHandler
from tweepy.streaming import StreamListener
import time

# Navigate to your twitter apps keys and use them.
ckey="aJ8N5zsMXnaRBJpJUBPfsVN2m"
csecret="mMBLEAw2YiaPST9EEXsLbKBw68LfzWQU5yiVB8FkTZn9a6B27c"
atoken="587454349-46cFBwggREZMn9CFOlAy0t9Arh68hWUDKbTuA44W"
asecret="BP8dC80EDGpJvackYNs6Qu6BJ9rShTLwHTB09z6XCZiIP"

class listener(StreamListener):
    def on_data(self, data):
        try:
            print(data)
            return(True)
        except BaseException as e:
            print("Network Error", str(e))
            time.sleep(2)
    def on_error(self, status):
        print(status)

auth = OAuthHandler(ckey, csecret)
auth.set_access_token(atoken, asecret)

twitterStream = Stream(auth, listener())
twitterStream.filter(track=["car"]) # track = ["this will give you relevant tweets as your word Here"]


{"created_at":"Sat Mar 03 17:36:34 +0000 2018","id":969989926170320896,"id_str":"969989926170320896","text":"RT @TrillxLove: I got apartment fever, car fever and better life fever . Y\u2019all can keep that baby fever I\u2019m tryna Be successful before I st\u2026","source":"\u003ca href=\"http:\/\/twitter.com\/download\/iphone\" rel=\"nofollow\"\u003eTwitter for iPhone\u003c\/a\u003e","truncated":false,"in_reply_to_status_id":null,"in_reply_to_status_id_str":null,"in_reply_to_user_id":null,"in_reply_to_user_id_str":null,"in_reply_to_screen_name":null,"user":{"id":519129349,"id_str":"519129349","name":"Dallas Cranford","screen_name":"Dallll_assss","location":"Asheboro, NC","url":null,"description":"8\/3\/15\ud83d\udc98","translator_type":"none","protected":false,"verified":false,"followers_count":1014,"friends_count":438,"listed_count":0,"favourites_count":20776,"statuses_count":13183,"created_at":"Fri Mar 09 03:10:57 +0000 2012","utc_offset":null,"time_zone":null,"geo_enabled":true,"l

{"created_at":"Sat Mar 03 17:36:34 +0000 2018","id":969989927097327616,"id_str":"969989927097327616","text":"RT @KimJongUnique: La S\u00e9cu rembourse les IVG de confort, mais d\u00e9rembourserait les c\u00e9sariennes. Tout cela est d'une logique! https:\/\/t.co\/Uo\u2026","source":"\u003ca href=\"http:\/\/twitter.com\" rel=\"nofollow\"\u003eTwitter Web Client\u003c\/a\u003e","truncated":false,"in_reply_to_status_id":null,"in_reply_to_status_id_str":null,"in_reply_to_user_id":null,"in_reply_to_user_id_str":null,"in_reply_to_screen_name":null,"user":{"id":179481483,"id_str":"179481483","name":"S\u00e9bastien Roussel","screen_name":"SebRou55el","location":"France","url":null,"description":"Patriote Fran\u00e7ais, anti-mondialisation, anti-EU, anti-gauchiste, pour le retour de la vraie France.","translator_type":"none","protected":false,"verified":false,"followers_count":429,"friends_count":401,"listed_count":25,"favourites_count":570,"statuses_count":42746,"created_at":"Tue Aug 17 11:53:

{"created_at":"Sat Mar 03 17:36:34 +0000 2018","id":969989927684526080,"id_str":"969989927684526080","text":"RT @NikBovain: I wish ya would\u2019ve put in the headline that an OFF DUTY COP CRASHED INTO HER b\/c he was driving at 94 mph killing her infant\u2026","source":"\u003ca href=\"http:\/\/twitter.com\/download\/iphone\" rel=\"nofollow\"\u003eTwitter for iPhone\u003c\/a\u003e","truncated":false,"in_reply_to_status_id":null,"in_reply_to_status_id_str":null,"in_reply_to_user_id":null,"in_reply_to_user_id_str":null,"in_reply_to_screen_name":null,"user":{"id":136495551,"id_str":"136495551","name":"No Scary Spice","screen_name":"itsMelBee","location":"New York, USA","url":"http:\/\/www.melbeephoto.com","description":"Photographer \ud83d\udcf8 Houston \u2708\ufe0f NYC","translator_type":"none","protected":false,"verified":false,"followers_count":948,"friends_count":800,"listed_count":9,"favourites_count":1946,"statuses_count":45155,"created_at":"Sat Apr 24 03:17:58 +0000 2010","utc_offs

{"created_at":"Sat Mar 03 17:36:35 +0000 2018","id":969989928028442624,"id_str":"969989928028442624","text":"RT @NikBovain: I wish ya would\u2019ve put in the headline that an OFF DUTY COP CRASHED INTO HER b\/c he was driving at 94 mph killing her infant\u2026","source":"\u003ca href=\"http:\/\/twitter.com\/download\/iphone\" rel=\"nofollow\"\u003eTwitter for iPhone\u003c\/a\u003e","truncated":false,"in_reply_to_status_id":null,"in_reply_to_status_id_str":null,"in_reply_to_user_id":null,"in_reply_to_user_id_str":null,"in_reply_to_screen_name":null,"user":{"id":297064680,"id_str":"297064680","name":"lourdes","screen_name":"Viridixna","location":"Houston | ASU19","url":null,"description":"I\u2019d throw roses around | #HeretoStay \ud83c\uddf2\ud83c\uddfd | Angelo State Nursing \ud83d\udc0f\ud83d\udc99\ud83d\udc9b","translator_type":"none","protected":false,"verified":false,"followers_count":1674,"friends_count":1640,"listed_count":23,"favourites_count":36296,"statuses_count":76059,"creat

{"created_at":"Sat Mar 03 17:36:35 +0000 2018","id":969989929886400513,"id_str":"969989929886400513","text":"RT @Scarlet_Pioneer: R.I.P. Porn Star Billy Herrington Died From Car Accident https:\/\/t.co\/haWEqL6T6n \n\n\u5143\u30dd\u30eb\u30ce\u30b9\u30bf\u30fc\u3001\u30d3\u30ea\u30fc\u30fb\u30d8\u30ea\u30f3\u30c8\u30f3\u3055\u3093\u304c\u4ea4\u901a\u4e8b\u6545\u306b\u3088\u308a\u6b7b\u4ea1\u3002\u4eab\u5e7448","source":"\u003ca href=\"http:\/\/twitter.com\/download\/android\" rel=\"nofollow\"\u003eTwitter for Android\u003c\/a\u003e","truncated":false,"in_reply_to_status_id":null,"in_reply_to_status_id_str":null,"in_reply_to_user_id":null,"in_reply_to_user_id_str":null,"in_reply_to_screen_name":null,"user":{"id":3791943853,"id_str":"3791943853","name":"\u3060\u3084\u307e@\u6709\u6fa4","screen_name":"kousuke092800","location":null,"url":null,"description":"\u592a\u3044\u65b9\u306e\u3060\u3084\u307e\u306f\u30b3\u30c1\u30e9\u306b\u306a\u308a\u307e\u3059\n\u304a\u3063\u3055\u3093\u3060\u305

{"created_at":"Sat Mar 03 17:36:35 +0000 2018","id":969989929919954944,"id_str":"969989929919954944","text":"RT @skymochisakura: \u30d3\u30ea\u30fc\u30d8\u30ea\u30f3\u30c8\u30f3\u6c0f\u304c\u4ea4\u901a\u4e8b\u6545\u3067\u304a\u4ea1\u304f\u306a\u308a\u306b...\nhttps:\/\/t.co\/kJ59aKlalS\n\u672a\u3060\u306b\u4fe1\u3058\u3089\u308c\u306a\u3044\u3093\u3060\u3051\u3069\n\u30fb\u30d8\u30ea\u30f3\u30c8\u30f3\u6c0f\u306e\u53cb\u4eba\u306e\u30a2\u30ab\u30a6\u30f3\u30c8\u3089\u304c\u8ffd\u60bc\u30c4\u30a4\u30fc\u30c8\n\u30fb\u4e0a\u8a18\u306e\u30b5\u30a4\u30c8\u304c\u6709\u540d\u306a\u30b2\u30a4\u30dd\u30eb\u30ce\u30b5\u30a4\u30c8\n\u30fb\u73fe\u306b\u672c\u4eba\u306e\u30c4\u30a4\u30fc\u30c8\u304c\u2026","source":"\u003ca href=\"https:\/\/mobile.twitter.com\" rel=\"nofollow\"\u003eTwitter Lite\u003c\/a\u003e","truncated":false,"in_reply_to_status_id":null,"in_reply_to_status_id_str":null,"in_reply_to_user_id":null,"in_reply_to_user_id_str":null,"in_reply_to_screen_name":null,"user":{"id":1232788

{"created_at":"Sat Mar 03 17:36:35 +0000 2018","id":969989930167537664,"id_str":"969989930167537664","text":"RT @_keetikat: I just know that red car turned the radio down &amp; rode in silence for the next 10 minutes. https:\/\/t.co\/0eoOcTGkIR","source":"\u003ca href=\"http:\/\/twitter.com\/download\/iphone\" rel=\"nofollow\"\u003eTwitter for iPhone\u003c\/a\u003e","truncated":false,"in_reply_to_status_id":null,"in_reply_to_status_id_str":null,"in_reply_to_user_id":null,"in_reply_to_user_id_str":null,"in_reply_to_screen_name":null,"user":{"id":354076988,"id_str":"354076988","name":"AT3\u20e3","screen_name":"thompson_0829","location":"1235 Phillip Fulmer Way ","url":"https:\/\/Instagram.com\/thompson_0829\/","description":"Atlanta Braves\u26be\ufe0f Tennessee Vols\ud83c\udf4a HTTR\ud83d\udc37 Give Him Six\u2122 Psalms 23:4\ud83d\ude4f\ud83c\udffcI tweet real shit and song lyrics \ud83e\udd37\ud83c\udffb\u200d\u2642\ufe0f X.VIII.XVII\ud83e\udde1 @MoMoGoodbrod is mine","translator_type":

{"created_at":"Sat Mar 03 17:36:35 +0000 2018","id":969989930859380736,"id_str":"969989930859380736","text":"RT @skymochisakura: \u30d3\u30ea\u30fc\u30d8\u30ea\u30f3\u30c8\u30f3\u6c0f\u304c\u4ea4\u901a\u4e8b\u6545\u3067\u304a\u4ea1\u304f\u306a\u308a\u306b...\nhttps:\/\/t.co\/kJ59aKlalS\n\u672a\u3060\u306b\u4fe1\u3058\u3089\u308c\u306a\u3044\u3093\u3060\u3051\u3069\n\u30fb\u30d8\u30ea\u30f3\u30c8\u30f3\u6c0f\u306e\u53cb\u4eba\u306e\u30a2\u30ab\u30a6\u30f3\u30c8\u3089\u304c\u8ffd\u60bc\u30c4\u30a4\u30fc\u30c8\n\u30fb\u4e0a\u8a18\u306e\u30b5\u30a4\u30c8\u304c\u6709\u540d\u306a\u30b2\u30a4\u30dd\u30eb\u30ce\u30b5\u30a4\u30c8\n\u30fb\u73fe\u306b\u672c\u4eba\u306e\u30c4\u30a4\u30fc\u30c8\u304c\u2026","source":"\u003ca href=\"http:\/\/twitter.com\/download\/iphone\" rel=\"nofollow\"\u003eTwitter for iPhone\u003c\/a\u003e","truncated":false,"in_reply_to_status_id":null,"in_reply_to_status_id_str":null,"in_reply_to_user_id":null,"in_reply_to_user_id_str":null,"in_reply_to_screen_name":null,"use

KeyboardInterrupt: 

** The error in last is due to interrupt I made with 'Ctrl+c' to terminate process. **
## Yeah, we just get the tweets from Twitter API. 

## But wait, what is the thing in above block are these tweets or garbage ?

![confusion](/img/conf.JPG)

## Ofcourse, these are tweets but with lot more information we don't need. 
** Lets perform third key point written in first para of this tutorial, "Text Pre-processing or cleaning the tweets." **

In [4]:
# Importing Libraries
from tweepy import Stream
from tweepy import OAuthHandler
from tweepy.streaming import StreamListener
import time

# Navigate to your twitter apps keys and use them.
ckey="aJ8N5zsMXnaRBJpJUBPfsVN2m"
csecret="mMBLEAw2YiaPST9EEXsLbKBw68LfzWQU5yiVB8FkTZn9a6B27c"
atoken="587454349-46cFBwggREZMn9CFOlAy0t9Arh68hWUDKbTuA44W"
asecret="BP8dC80EDGpJvackYNs6Qu6BJ9rShTLwHTB09z6XCZiIP"

class listener(StreamListener):

    def on_data(self, data):
        try:
            tweet = data.split(',"text":"')[1].split('","source')[0]
            print (tweet)
            return(True)
        except BaseException as e:
            print ("Network Error", str(e))
            time.sleep(2)
    def on_error(self, status):
        print (status)

auth = OAuthHandler(ckey, csecret)
auth.set_access_token(atoken, asecret)

twitterStream = Stream(auth, listener())
twitterStream.filter(track=["car"])

RT @Steve_Pippin: @BelBlok \n@BrandonBEP1 \n@busylizzie48 \n@DeadofKnight68 \n@Jeanniespeaches \n@mandytrumpteam \n@nerbonnej\n@OneNation1G \n@Patr\u2026
RT @taka__0406: \u5144\u8cb4\u81ea\u52d5\u8eca\u4e8b\u6545\u3067\u4ea1\u304f\u306a\u3063\u3066\u305f\u306e\u304b\u2026\n\u4fe1\u3058\u3089\u308c\u306a\u3044\nhttps:\/\/t.co\/kZiLRysjEB
Patinodromo del Espinal estrena sistema de iluminaci\u00f3n, la pista espera realizar eventos de car\u00e1cter nacional con l\u2026 https:\/\/t.co\/iOwVx1mdZ2","display_text_range":[0,140],"source":"\u003ca href=\"http:\/\/twitter.com\/download\/android\" rel=\"nofollow\"\u003eTwitter for Android\u003c\/a\u003e","truncated":true,"in_reply_to_status_id":null,"in_reply_to_status_id_str":null,"in_reply_to_user_id":null,"in_reply_to_user_id_str":null,"in_reply_to_screen_name":null,"user":{"id":389267343,"id_str":"389267343","name":"Chelo Delgado","screen_name":"chelodelgado89","location":null,"url":null,"description":"Periodista Deportivo \/ Comunicador Soc

RT @NikBovain: I wish ya would\u2019ve put in the headline that an OFF DUTY COP CRASHED INTO HER b\/c he was driving at 94 mph killing her infant\u2026
Exactamente por eso estoy sola, por eso no soy la favorita de muchas personas. \nPorque tengo un car\u00e1cter fuerte, un\u2026 https:\/\/t.co\/ZrF9D3qHU8
Also- FREE CAR WASH \ud83d\ude01\ud83d\ude4c\ud83c\udffc #LARain
RT @willdriskill: Did a car just flash their lights at me or hit a bump?? Questions I always ask myself while driving
RT @NikBovain: I wish ya would\u2019ve put in the headline that an OFF DUTY COP CRASHED INTO HER b\/c he was driving at 94 mph killing her infant\u2026
RT @Steve_Pippin: ALL NEW on the #PatriotsEffectingChange Train!\n \n@loolo1022\n@USACitizen7\n@cjane53\n@Melissa1opinion\n@Makiti87\n@unctarheels\u2026
RT @LopezVargasR: #GNBInforma | Efectivos @GNB_BARINAS detuvieron 1 Cddno. y retuvieron 1 veh\u00edculo y 10 TONELADAS de Ca\u00f1a de Az\u00facar por des\u2026
RT @Scarlet_Pioneer: R.I.P. Porn Star Billy 

RT @NikBovain: I wish ya would\u2019ve put in the headline that an OFF DUTY COP CRASHED INTO HER b\/c he was driving at 94 mph killing her infant\u2026
STOLEN Female Cocker Spaniel  #Heathfield #EastSussex on 1.3.18 May have been dumped as was not with the younger bi\u2026 https:\/\/t.co\/hIowG7tjAJ","display_text_range":[0,140],"source":"\u003ca href=\"http:\/\/twitter.com\/#!\/download\/ipad\" rel=\"nofollow\"\u003eTwitter for iPad\u003c\/a\u003e","truncated":true,"in_reply_to_status_id":null,"in_reply_to_status_id_str":null,"in_reply_to_user_id":null,"in_reply_to_user_id_str":null,"in_reply_to_screen_name":null,"user":{"id":2786181192,"id_str":"2786181192","name":"Stolen&Missing Pets","screen_name":"SAMPAuk_","location":"UK","url":"http:\/\/www.stolenandmissingpetsalliance.co.uk","description":"Tougher pet theft legislation; Re-classification of dog theft; Compulsory microchip scanning & proof of ownership through a chip;","translator_type":"none","protected":false,"verified":false,

If there\u2019s a gofundme for her lawyer fees, please pass it along. She deserves the best and that cop should rot in p\u2026 https:\/\/t.co\/C2WCvRZxq5","display_text_range":[0,140],"source":"\u003ca href=\"http:\/\/twitter.com\/download\/iphone\" rel=\"nofollow\"\u003eTwitter for iPhone\u003c\/a\u003e","truncated":true,"in_reply_to_status_id":null,"in_reply_to_status_id_str":null,"in_reply_to_user_id":null,"in_reply_to_user_id_str":null,"in_reply_to_screen_name":null,"user":{"id":3270196838,"id_str":"3270196838","name":"Love, Rae \ud83d\ude45\u2600\ufe0f","screen_name":"RaezillaHockey","location":"Colorado, USA","url":null,"description":"NHL nonsense. Mostly, Caps, Devils, Avs and Jets - also a whole lot of antiracist resisterhood blabbing. Take me or leave me \ud83e\udd37\u200d\u2640\ufe0f\u26a1\ufe0f\ud83c\udf08\ud83d\udcab #nanowrimo #amwriting","translator_type":"none","protected":false,"verified":false,"followers_count":902,"friends_count":750,"listed_count":40,"favourites_coun

RT @egg_dog: a car is a metal ravioli and you are the meat!
RT @GordonMercedes9: Car Parked Darling but the Kitchen's a mess\n\n@EastEndGentxxx @fokkerdude @Darylmatla @ProfessorTeresa @alfaqfour @Cosi\u2026
RT @NikBovain: I wish ya would\u2019ve put in the headline that an OFF DUTY COP CRASHED INTO HER b\/c he was driving at 94 mph killing her infant\u2026
RT @The_Animal_Team: So sorry to say Piper was found by one of the search team not far from where she went missing and early thoughts are s\u2026
RT @TeaPainUSA: THEN: Former Attorney General Janet Reno once paid full price for a new car so no one would think she used her office to ge\u2026
RT @NikBovain: I wish ya would\u2019ve put in the headline that an OFF DUTY COP CRASHED INTO HER b\/c he was driving at 94 mph killing her infant\u2026
RT @NikBovain: I wish ya would\u2019ve put in the headline that an OFF DUTY COP CRASHED INTO HER b\/c he was driving at 94 mph killing her infant\u2026
@brevesdejours @_Math_D69 @LeLapinTaquin ...

KeyboardInterrupt: 

## Yes, now we can see some fine tweets, but still there are some non relevant info but really it's ok.. We can delete them manually.

# Caveat! "Donot run Python code to extract tweets for more than 24 hours, It is voilation of Twitter Apps Agreement".

## Let's save these tweets in file in our directory, for sentiment analysis.

In [9]:
# Importing Libraries
from tweepy import Stream
from tweepy import OAuthHandler
from tweepy.streaming import StreamListener
import time

# Navigate to your twitter apps keys and use them.
ckey="aJ8N5zsMXnaRBJpJUBPfsVN2m"
csecret="mMBLEAw2YiaPST9EEXsLbKBw68LfzWQU5yiVB8FkTZn9a6B27c"
atoken="587454349-46cFBwggREZMn9CFOlAy0t9Arh68hWUDKbTuA44W"
asecret="BP8dC80EDGpJvackYNs6Qu6BJ9rShTLwHTB09z6XCZiIP"

class listener(StreamListener):

    def on_data(self, data):
        try:
            tweet = data.split(',"text":"')[1].split('","source')[0]
            print (tweet)
            tweet = str(tweet)
            with open("tweets.txt", "a") as f:
                f.write(tweet)
                f.write("\n")
            return(True)
        except BaseException as e:
            print ("Network Error", str(e))
            time.sleep(2)
    def on_error(self, status):
        print (status)

auth = OAuthHandler(ckey, csecret)
auth.set_access_token(atoken, asecret)

twitterStream = Stream(auth, listener())
twitterStream.filter(track=["car"])

RT @TitusNation: Wilbur Ross, Trumps Steel Factory buddy said steel tariffs would add 1% to the cost of a car. \nOK. \n40k car, that is only\u2026
RT @NikBovain: I wish ya would\u2019ve put in the headline that an OFF DUTY COP CRASHED INTO HER b\/c he was driving at 94 mph killing her infant\u2026
RT @Itslilstacy: Niggas in henry co stay in cribs like this &amp; swear they from the hood. Foh\ud83d\ude02\ud83d\ude02\ud83d\udc4e\ud83c\udffe https:\/\/t.co\/ApH7cjGL6B
#trixie #instagramers #instalike #instagram #instagood #roadslikethese #life #sunset #cars #car\u2026 https:\/\/t.co\/yd3mosADVl
@Drawing85796888 This one reminds me of the car lines ordering fast food; I often just go inside.","display_text_range":[17,97],"source":"\u003ca href=\"http:\/\/twitter.com\" rel=\"nofollow\"\u003eTwitter Web Client\u003c\/a\u003e","truncated":false,"in_reply_to_status_id":967428224211808256,"in_reply_to_status_id_str":"967428224211808256","in_reply_to_user_id":933578565043437568,"in_reply_to_user

KeyboardInterrupt: 

** Now, most of the tweets from API are pre-processed and saved in a text file. **
![file](/img/file.JPG)

## Finally, we are about to perform sentiment analysis with 'textblob'.

** TextBlob is a Python (2 and 3) library for processing textual data. It provides a simple API for diving into common natural language processing (NLP) tasks such as part-of-speech tagging, noun phrase extraction, sentiment analysis, classification, translation, and more. **

** download textblob as follows,**
* open command prompt in windows or teminal in Linux/MAC.
* .>>> pip install -U textblob
* .>>> python -m textblob.download_corpora 

last line written above will download 'Bag of Words', to determine the relevant and irrelevant words.

Use link to read more about:
1) [Bag of Words](https://en.wikipedia.org/wiki/Bag-of-words_model)
2) [textblob Documentation](http://textblob.readthedocs.io/en/dev/)

In [25]:
from textblob import TextBlob
i = 0

def read_data():
    tweets = open("tweets.txt", "r").readlines()
    return tweets
def Sentiment_analysis(tweet):
    analysis = TextBlob(tweet)
    return (analysis.sentiment)

tweets = read_data()


for tweet in tweets:
    print(tweet)
    analysis = Sentiment_analysis(tweet)
    print(analysis)
    print("\n"+"------")
    i += 1
    if i == 2:
        break

this car is really fast and cool.

Sentiment(polarity=0.275, subjectivity=0.625)

------
I seriously dont like this car.

Sentiment(polarity=-0.3333333333333333, subjectivity=0.6666666666666666)

------


** Sentiment(polarity=x.xx, subjectivity=x.xx), in this line 'polarity' shows how good or bad the sentence is, and 'subjectivity' shows that the reality of sentence as per universe. **

** As, you can that first tweet seems to be good thus (polarity = 0.275) and second tweet seems to be bad that's why (polarity = -0.33) **

### And, guess what ! we have just completed the sentiment analysis of Tweets from Twitter Apps API in Python and Machine Learning. 

### As, textblob is Machine Learning easiest way to start sentiment analysis for text. It is by default trained with 'corpora' we downloaded before. 

## Hope you enjoyed the tutorial, I know, I know you guys need to know more about sentiment analysis but you must have good knowledge of NLP (Natural Language Processing).
 
### For NLP, I will write a complete guidance blog soon.

##  But, for now Congratulations, and start analyzing the sentiments of your tweets.

![congratz](/img/congo.JPG)

## Topic for next tutorial is not decided yet, you can comment in video or directly ping me for the topic you want.

## Check my other tutorials as well, and also give your reviews on my research papers.

* [More Tutorials on GitHub](https://github.com/elacsoft)
* [Research Papers](https://github.com/elacsoft/Research-Papers)

## Syed Owais Chishti
### Data Scientist.

>* [MyWebsite](http://elacsoft.cf) 
* [MyTwitterAccount](https://twitter.com/robertw26984557)
* [MyFacebookAccount](https://www.facebook.com/owais.chishti.35)
* [MyGitHubAccount](https://github.com/elacsoft)
* [MyYoutubeChannel](https://www.youtube.com/channel/UCWha7dSnT_k7mBGJ0XFB0IQ)
* [elacsoft@gmail.com](elacsoft@gmail.com)